pkg/network/server.go

```diff
@@ -347,13 +348,28 @@ func (s *Server) HandshakedPeersCount() int {
 
 // getVersionMsg returns current version message.
 func (s *Server) getVersionMsg() *Message {
+	capabilities := []capability.Capability{
+		{
+			Type: capability.FullNode,
+			Data: &capability.Node{
+				StartHeight: s.chain.BlockHeight(),
+			},
+		},
+	}
+	if s.Port > 0 {
```

Comment:
Technically we'll have some port being used anyway and we probably should be using that, but only `s.transport` actually knows about it, maybe we should ask? See [`Listen` documentation](https://golang.org/pkg/net/#Listen) for port number zero handling. I think it will make us a bit more flexible and will allow us to test the node properly in the future (the same we run RCP server on dynamic port at the moment during the test).

Reply:
And I think this capability should always be present for our case.

Reply:
Added `Address()` method to `Transporter` interface to be sure about the port. This also requires more complicated test transporter to fill TCP capability correctly.